analyze_issue_metadata.py

```diff
@@ -0,0 +1,186 @@
+
+    Returns:
+        A JSON of issue metadata
+
+    Raises:
+        N/A
+    """
+
+    org = str(sys.argv[1])
+
+    with open(sys.argv[2]) as f:
+        repos = f.readlines()
+    repos = [x.strip() for x in repos]
+
+    # Authenticate with GitHub using Personal Access Token
+    g = github3.login(token=os.environ['GH_TOKEN'])
+
+    data = gim.get_issue_metadata(g, org, repos, state)
+
+    # Dump the data to a JSON file
+    data_json = json.dumps(data)
+    file_time = dt.now().strftime("%Y%m%d")
+    print 'Dumping data to json file'
+    with open('../output_files/' + file_time + '_raw_veneer_issue_meta.json',
+              'w') as f:
+        data = json.dump(data, f, sort_keys=True, indent=4, separators=(
+             ',', ': '))
+
+    return data_json
+
+
+def analyze_issue_metadata(data):
+
+    p0_labels = ['Priority: P0', 'priority: P0', 'priority: p0', 'P0', 'p0']
+    p1_labels = ['Priority: P1', 'priority: P1', 'priority: p1', 'P1', 'p1']
+    p2_labels = ['Priority: P2+', 'priority: P2+', 'priority: p2+',
+                 'P2+', 'p2+', 'Priority: P2', 'priority: P2', 'priority: p2',
+                 'P2', 'p2']
+    fr_question = ['Type: Enhancement', 'Type: Question', 'type: enhancement',
+                   'type: question', 'enhancement']
+
+    analysis = {}
+
+    # Load/decode the data JSON
+    data_decode = json.loads(data)
+
+    for repo in data_decode:
+        if repo == 'updated':
+            continue
+        else:
+
+            analysis[repo] = {'issues': {'count': 0, 'age': 0},
+                              'prs': {'count': 0, 'age': 0},
+                              'p0': {'count': 0, 'age': 0},
+                              'p1': {'count': 0, 'age': 0},
+                              'p2+': {'count': 0, 'age': 0},
+                              'no_priority_label': {'count': 0, 'age': 0},
+                              'fr_question': {'count': 0, 'age': 0}}
+
+            analysis[repo]['issues']['count'] = data_decode[repo][
+                                                  'open_issues_count']
+            analysis[repo]['prs']['count'] = data_decode[repo]['prs'][
+                                               'open_pr_count']
+
+            # if there are prs, calculate their mean age
+            if analysis[repo]['prs']['count'] > 0:
+                analysis[repo]['prs']['age'] = data_decode[repo]['prs'][
+                    'pr_aggregate_age']/analysis[repo]['prs']['count']
+
+            for issue in data_decode[repo]['issues']:
+                created_date = dt.strptime(
+                    data_decode[repo]['issues'][issue]['created'],
+                    "%Y-%m-%dT%H:%M:%S+00:00")
+
+                # it is invalid for an issue to carry multiple priority labels
+                # so count the most severe p0 THEN p1 THEN p2+ THEN fr_question
+
+                if any(x in data_decode[repo]['issues'][issue]['labels']
```

Comment:
I'm concerned about the dependency on alphabetical ordering of the labels.  I would like to capture 'enhancement' as 'fr_question' for example.  

In an earlier version I used regex for all the labels but abandoned it because I couldn't see past a loop problem.  Your solution - the function - solves for that I think so maybe re.search on each label in the list would do the trick?  Wouldn't have to sort the list - just do the if statements in the order of severity: e.g. p1 trumps enhancement if both labels are present.